### PR TITLE
Correctly merge separate arrow arrays offsets to one chunk offset array

### PR DESCRIPTION
**Pull request by fexolm**

### CODE REVIEW

**Comment by fexolm** [comment]
@anton-malakhov PTAL

File: DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp [hunk 1]
```diff
@@ -32,24 +32,43 @@ void ArrowCsvForeignStorage::read(const ChunkKey& chunk_key,
   std::array<int, 3> col_key {chunk_key[0], chunk_key[1], chunk_key[2]};
   auto &frag = g_columns.at(col_key).at(chunk_key[3]);
   int64_t sz, copied = 0;
+  std::shared_ptr<arrow::ArrayData> prev_data = nullptr;
```
Comment (anton-malakhov) [inline DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp:35]: operating shared pointers is expensive excessive op when no ownership guarantee is required
```suggestion
  arrow::ArrayData* prev_data = nullptr;
```

File: DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp [hunk 2]
```diff
@@ -58,6 +77,7 @@ void ArrowCsvForeignStorage::read(const ChunkKey& chunk_key,
       else CHECK(false); // TODO: else???
     }
     dest += sz; copied += sz;
+    prev_data = array_data;
```
Comment (anton-malakhov) [inline DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp:80]:
```suggestion
    prev_data = array_data.get();
```

File: DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp [hunk 3]
```diff
@@ -32,24 +32,43 @@ void ArrowCsvForeignStorage::read(const ChunkKey& chunk_key,
...
+        // We are trying to merge separate arrow arrays with offsets to one omnisci chunk
+        // So we need to know that:
+        // - offsets length of every array = array.len() + 1
+        // - offsets array of every array starts from zero
+        // So, we calculate offset of (n+1)th array as a sum of last (max) offsets of previous n arrays
+        // and ignore first value of every new array offset
+        if(prev_data != nullptr && sz > 0) {
+          data++;
+          sz-=sizeof(uint32_t);
+        }
```
Comment (anton-malakhov) [inline DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp:62]: I don't quite understand this branch.. is the first element is special? why it is not skipped for the first chunk being copied? is it explained in the comment? also, if the prev_data pointer is just a flag - would it be more efficient to make it so?

**Pull request merged**